engine/src/test/java/org/camunda/bpm/engine/impl/batch/BatchInvocationsPerJobByBatchTypeTest.java

```diff
@@ -0,0 +1,125 @@
+import org.camunda.bpm.engine.runtime.ProcessInstanceQuery;
+import org.camunda.bpm.engine.test.Deployment;
+import org.camunda.bpm.engine.test.RequiredHistoryLevel;
+import org.camunda.bpm.engine.test.util.ProvidedProcessEngineRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class BatchInvocationsPerJobByBatchTypeTest {
+
+  @Rule
+  public ProvidedProcessEngineRule processEngineRule = new ProvidedProcessEngineRule();
+
+  protected ManagementService managementService;
+  protected RuntimeService runtimeService;
+  protected HistoryService historyService;
+  protected ProcessEngineConfigurationImpl engineConfiguration;
+
+  @Before
+  public void assignServices() {
+    managementService = processEngineRule.getManagementService();
+    runtimeService = processEngineRule.getRuntimeService();
+    historyService = processEngineRule.getHistoryService();
+    engineConfiguration = processEngineRule.getProcessEngineConfiguration();
+  }
+
+  @After
+  public void tearDown() {
+    int defaultInvocationsPerJob =
+        ProcessEngineConfigurationImpl.DEFAULT_INVOCATIONS_PER_BATCH_JOB;
+
+    engineConfiguration
+        .setInvocationsPerBatchJobByBatchType(new HashMap<>())
+        .setInvocationsPerBatchJob(defaultInvocationsPerJob);
+
+    managementService.createBatchQuery()
+        .list()
+        .forEach(this::cascadeDeleteBatch);
+  }
+
+  @Test
+  @Deployment(resources = "org/camunda/bpm/engine/test/api/oneTaskProcess.bpmn20.xml")
+  public void shouldFallbackToDefaultConfigurationWhenBatchTypeIsNotConfigured() {
+    // given
+    runtimeService.startProcessInstanceByKey("oneTaskProcess");
+
+    ProcessInstanceQuery processInstanceQuery = runtimeService.createProcessInstanceQuery();
+
+    // when
+    Batch batchOne = runtimeService.deleteProcessInstancesAsync(processInstanceQuery, "");
+
+    // then
+    assertThat(batchOne.getInvocationsPerBatchJob()).isEqualTo(1);
+  }
+
+  @Test
+  @RequiredHistoryLevel(ProcessEngineConfiguration.HISTORY_ACTIVITY)
+  @Deployment(resources = "org/camunda/bpm/engine/test/api/oneTaskProcess.bpmn20.xml")
+  public void shouldFallbackToGlobalConfigurationWhenBatchTypeIsNotConfigured() {
+    // given
+    Map<String, Integer> invocationsPerBatchJobByBatchType =
+        Collections.singletonMap(Batch.TYPE_PROCESS_INSTANCE_DELETION, 10);
+
+    engineConfiguration
+        .setInvocationsPerBatchJobByBatchType(invocationsPerBatchJobByBatchType)
+        .setInvocationsPerBatchJob(42);
+
+    runtimeService.startProcessInstanceByKey("oneTaskProcess");
+
+    ProcessInstanceQuery processInstanceQuery = runtimeService.createProcessInstanceQuery();
+
+    Batch batchOne =
+        runtimeService.deleteProcessInstancesAsync(processInstanceQuery, "");
```

Comment:
Is this necessary in order to test what the test case title indicates? Should this rather be part of a second test case 'shouldUseConfiguredInvocationsForBatchType'?

Reply:
What we already cover by other tests:
* invocations by specific batch type
* global configuration (by older test cases already present in the code base)

However, I think it makes sense to have a scenario in which we assert that a mixed configuration works as expected. I'm open for proposals how to rename this test case.

Reply:
I totally agree with all points. 👍 
Leaves the question: is it technically necessary to start a process instance here and such, or is it enough to simply create the second batch and assert its configuration?

Reply:
The creation of this batch type would fail if no process instance is available [1]

[1] https://github.com/camunda/camunda-bpm-platform/blob/master/engine/src/main/java/org/camunda/bpm/engine/impl/cmd/batch/DeleteProcessInstanceBatchCmd.java#L73

Reply:
OK, I see, thanks for the clarification! 
The test case title is fine then as well. 👍